openlibrary/plugins/openlibrary/js/Browser.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Convert url paramaters to an object
+ * @param {String} urlSearch everything after (and including) the '?' of the url
+ * @returns {{ [param: string]: string }}
+ */
+export function getJsonFromUrl(urlSearch) {
+    const query = urlSearch.substr(1);
+    const result = {};
+    if (query) {
+        query.split('&').forEach(part => {
+            const item = part.split('=');
+            result[item[0]] = decodeURIComponent(item[1]);
+        });
+    }
+    return result;
+}
+
+/**
+ * @param {String} url
+ * @param {String} parameter name of param to remove
+ * @returns {String}
+ */
+export function removeURLParameter(url, parameter) {
+    var urlparts = url.split('?');
+    var prefix = urlparts[0];
+    var query, paramPrefix, params, i;
+    if (urlparts.length >= 2) {
+        query = urlparts[1];
+        paramPrefix = `${encodeURIComponent(parameter)}=`;
+        params = query.split(/[&;]/g);
+
+        //reverse iteration as may be destructive
+        for (i = params.length; i-- > 0;) {
+            //idiom for string.startsWith
+            if (params[i].lastIndexOf(paramPrefix, 0) !== -1) {
+                params.splice(i, 1);
+            }
+        }
+
+        url = prefix + (params.length > 0 ? `?${params.join('&')}` : '');
+        return url;
```

Comment:
@jdlrobson : combine above 2 lines?

Reply:
Actually I kind of find this easier to read; 

```js
return prefix + (params.length > 0 ? `?${params.join('&')}` : '');
```

Makes me think "Wait, what is this returning?", whereas

```js
url = prefix + (params.length > 0 ? `?${params.join('&')}` : '');
return url;
```

I can see "Ah, it's doing something to the url and then returning it."

So Won't fix.